taichi/ir/frontend_ir.h

```diff
@@ -408,11 +408,30 @@ class ExternalTensorExpression : public Expression {
   int element_dim;  // 0: scalar; 1: vector (SOA); 2: matrix (SOA); -1: vector
                     // (AOS); -2: matrix (AOS)
 
+  // Fill element shape if compile-time specialization is desired.
+  std::vector<int> element_shape;
+
   ExternalTensorExpression(const DataType &dt,
                            int dim,
                            int arg_id,
                            int element_dim)
-      : dt(dt), dim(dim), arg_id(arg_id), element_dim(element_dim) {
+      : ExternalTensorExpression(dt,
```

Comment:
Manual initialization for std::vector isn't needed so a better way to write this is 
```
  ExternalTensorExpression(const DataType &dt,
                           int dim,
                           int arg_id,
                           int element_dim):
      : dt(dt),
        dim(dim),
        arg_id(arg_id),
        element_dim(element_dim) {
    set_attribute("dim", std::to_string(dim));
}
  ExternalTensorExpression(const DataType &dt,
                           int dim,
                           int arg_id,
                           int element_dim,
                           const std::vector<int> &element_shape) :
 ExternalTensorExpression(dt,dim, arg_id, element_dim), element_shape(element_shape) {}
```

Reply:
Well the compiler complains "a delegating constructor cannot have other mem-initializers".

I can just leave the std vector a blank initializer

Reply:
oh sorry so 
```
ExternalTensorExpression(dt,dim, arg_id, element_dim) {this->element_shape = element_shape;}
```
should work?

Reply:
Ah this should work, but does it look better than mem initializers?

Reply:
Looks better actually :)

Thanks!